### PR TITLE
Hard-code version number value for CBlockLocator and CDiskBlockIndex

### PR DESCRIPTION
**Pull request by theuni**
This is also a much simpler replacement for #28327.

There are version fields in `CBlockLocator` and `CDiskBlockIndex` that have always been written but discarded when read.

I intended to convert them to use SerParams as introduced by #25284, which [ended up looking like this](https://github.com/theuni/bitcoin/commit/3e3af451652322c92e8e41cf918e69d608ec7c77). However because we don't currently have any definition of what a hash value would mean for either one of those, and we've never assigned the version field any meaning, I think it's better to just not worry about them.

If we ever need to assign meaning in the future, we can introduce `SerParams` as was done for `CAddress`.

As for the dummy values chosen:

`CDiskBlockIndex::DUMMY_VERSION` was easy as the highest ever client version, and I don't expect any objection there.

`CBlockLocator::DUMMY_VERSION` is hard-coded to the higest _PROTOCOL_ version ever used. This is to avoid a sudden bump that would be visible on the network if CLIENT_VERSION were used instead. In the future, if we ever need to use the value, we can discard anything in the CLIENT_VERSION range (for a few years as needed), as it's quite a bit higher.

While reviewing, I suggest looking at the throwaway `SerParams` commit above as it shows where the call-sites are. I believe that should be enough to convince one's self that hashing is never used.

### CODE REVIEW

**Comment by theuni** [comment]
Ping @sipa @MarcoFalke

**Comment by ajtowns** [comment]
utACK c1aa0af69eee05a9fad1eabd45662fc2143a3e0c

File: src/chain.h [hunk 1]
```diff
@@ -388,6 +388,13 @@ const CBlockIndex* LastCommonAncestor(const CBlockIndex* pa, const CBlockIndex*
 /** Used to marshal pointers into hashes for db storage. */
 class CDiskBlockIndex : public CBlockIndex
 {
+    /** Historically CBlockLocator's version field has been written to disk
+     * streams as the client version, but the value has never been used.
+     *
+     * Hard-code to the highest client version ever written.
+     * SerParams can be used if the field requires any meaning in the future.
+    **/
```
Comment (TheCharlatan) [inline src/chain.h:396]: Nitty nit: My `clang-format-diff` wants this to align with the stars in the line above.

**Review by sedited — APPROVED** [review]
ACK c1aa0af69eee05a9fad1eabd45662fc2143a3e0c

**Comment by theuni** [comment]
Resolved formatting nit.

Reviewers can confirm locally that the diff from before amounts only to whitespace:
`git diff -w --ignore-blank-lines c1aa0af69eee05a9fad1eabd45662fc2143a3e0c..e73d2a8018def940afadb5d699b18f39e882c1fc`

**Comment by ajtowns** [comment]
reACK e73d2a8018def940afadb5d699b18f39e882c1fc

File: src/dbwrapper.h [hunk 2]
```diff
@@ -167,7 +166,7 @@ class CDBIterator
 
     template<typename V> bool GetValue(V& value) {
         try {
-            CDataStream ssValue{GetValueImpl(), SER_DISK, CLIENT_VERSION};
+            DataStream ssValue{GetValueImpl()};
```
Comment (TheCharlatan) [inline src/dbwrapper.h:169]: Just saw that this did replace the `CDataStream` with a `DataStream` here in `CDBIterator` and `CDBWrapper`, but it is still using a `CDataStream` for `CDBBatch`. Should that be replaced too, as I've done here: https://github.com/TheCharlatan/bitcoin/commit/95405b8a4f3816277c3dbf075c8da013226e6cec?

Comment (theuni) [reply src/dbwrapper.h:169]: Yes please, thanks for catching this. IIRC I couldn't do this with my original approach of converting to `SerParams`, but forgot about it after switching. As @MarcoFalke says, this must be correct if it compiles.

**Pull request merged**